api/src/main/kotlin/edu/wgu/osmt/elasticsearch/OsmtQueryHelper.kt

```diff
@@ -0,0 +1,121 @@
+            log.debug(String.Companion.format("\n%s dslFilter:\n\t\t%s", msgPrefix, nativeQuery.filter.toString()))
+        }
+        if (nativeQuery.query != null && log != null) {
+            log.debug(String.Companion.format("\n%s query:\n\t\t%s", msgPrefix, nativeQuery.query.toString()))
+        }
+    }
+
+    fun createMatchPhrasePrefixDslQuery(fieldName: String, searchStr: String, boostVal : Float? = null): co.elastic.clients.elasticsearch._types.query_dsl.Query {
+        return QueryBuilders.matchPhrasePrefix { qb -> qb.field(fieldName).query(searchStr).boost(boostVal) }
+    }
+
+    fun createMatchBoolPrefixDslQuery(fieldName: String, searchStr: String, boostVal : Float? = null): co.elastic.clients.elasticsearch._types.query_dsl.Query {
+        return QueryBuilders.matchBoolPrefix { qb -> qb.field(fieldName).query(searchStr).boost(boostVal) }
+    }
+
+    fun createSimpleQueryDslQuery(fieldName: String, searchStr: String, boostVal : Float? = null): co.elastic.clients.elasticsearch._types.query_dsl.Query {
+        return QueryBuilders.simpleQueryString { qb ->
+            qb.fields(fieldName).query(searchStr).boost(boostVal).defaultOperator(Operator.And)
+        }
+    }
+
+    fun createNestedQueryDslQuery(path: String, scoreMode: ChildScoreMode, query: co.elastic.clients.elasticsearch._types.query_dsl.Query? = null, innerHits: InnerHits? = null): co.elastic.clients.elasticsearch._types.query_dsl.Query {
+        query ?: QueryBuilders.matchAll { b -> b }
+        innerHits ?: InnerHits.Builder().build()
+        return QueryBuilders.nested { qb ->
+            qb.path(path)
+                .scoreMode(ChildScoreMode.Avg)
+                .innerHits(innerHits)
+                .query(QueryBuilders.matchAll { b -> b })
+        }
+    }
+
+    fun createTermsDslQuery(fieldName: String, filterValues: List<String>, andFlag: Boolean = true): co.elastic.clients.elasticsearch._types.query_dsl.Query {
+        val values = filterValues
+            .stream()
+            .map { FieldValue.of(it) }
+            .collect(Collectors.toList())
+        val tqf = TermsQueryField.Builder()
+            .value(values)
+            .build()
+        val terms = QueryBuilders.terms()
+            .field(fieldName)
+            .terms(tqf)
+            .build()
+            ._toQuery()
+        /* Short hand version https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/8.10/searching.html
+        val terms2 = terms { qb -> qb.field(fieldName).terms(tqf) }
+        return bool { qb -> if (andFlag)
+                                qb.must(terms2)
+                            else
+                                qb.should(terms2) }
+        */
+
+        return QueryBuilders.bool()
+            .let {
+                if (andFlag) it.must(terms)
+                else         it.should(terms) }
+            .build()
+            ._toQuery()
+    }
+
+    // TODO handle case sensitivity
```

Comment:
@shnooga this is the likely reason for your API test failures. OSMT expect search results to be sorted case insensitive.